agent/collector.go

```diff
@@ -169,79 +175,138 @@ func (l *Collector) postMessage(endpoint string, m model.MessageBody) {
 	if err != nil {
 		log.Errorf("Unable to encode message: %s", err)
 	}
-	l.cfg.APIEndpoint.Path = endpoint
-	url := l.cfg.APIEndpoint.String()
+
+	responses := make(chan postResponse)
+	for _, ep := range l.cfg.APIEndpoints {
+		go l.postToAPI(ep, checkPath, body, responses)
+	}
+
+	// Wait for all responses to come back before moving on.
+	statuses := make([]*model.CollectorStatus, 0, len(l.cfg.APIEndpoints))
+	for i := 0; i < len(l.cfg.APIEndpoints); i++ {
+		url := l.cfg.APIEndpoints[i].Endpoint.String()
+		res := <-responses
+		if res.err != nil {
+			log.Error(res.err)
+			continue
+		}
+
+		r := res.msg
+		switch r.Header.Type {
+		case model.TypeResCollector:
+			rm := r.Body.(*model.ResCollector)
+			if len(rm.Message) > 0 {
+				log.Errorf("error in response from %s: %s", url, rm.Message)
+			} else {
+				statuses = append(statuses, rm.Status)
+			}
+		default:
+			log.Errorf("unexpected response type from %s: %d", url, r.Header.Type)
+		}
+	}
+
+	if len(statuses) > 0 {
+		l.updateStatus(statuses)
+	}
+}
+
+func (l *Collector) updateStatus(statuses []*model.CollectorStatus) {
+	curEnabled := atomic.LoadInt64(&l.realTimeEnabled) == 1
+
+	// If any of the endpoints wants real-time we'll do that.
+	var nextEnable int
+	for _, s := range statuses {
+		if s.ActiveClients > 0 && l.cfg.AllowRealTime {
+			if !curEnabled {
+				nextEnable = 1
+			} else {
+				nextEnable = 0
+			}
+			break
+		} else if s.ActiveClients == 0 && curEnabled {
+			nextEnable = -1
+		}
+	}
+	switch nextEnable {
+	case 1:
+		log.Info("Detected active clients, enabling real-time mode")
+		atomic.StoreInt64(&l.realTimeEnabled, 1)
+	case -1:
+		log.Info("Detected 0 clients, disabling real-time mode")
+		atomic.StoreInt64(&l.realTimeEnabled, 0)
+	}
+
+	// We will pick the maximum interval given since generally this is
+	// only set if we're trying to limit load on the backend.
+	maxInterval := 0 * time.Second
```

Comment:
You could simplify this function by throwing this `maxInterval` calculation in the above loop? Feel free to ignore if you think it won't actually simplify things